src/Microsoft.ComponentDetection.Common/Telemetry/Records/BcdeExecutionTelemetryRecord.cs

```diff
@@ -22,13 +25,16 @@ public class BcdeExecutionTelemetryRecord : BaseDetectionTelemetryRecord
 
         public string AgentOSDescription { get; set; }
 
-        public static TReturn Track<TReturn>(Func<BcdeExecutionTelemetryRecord, TReturn> functionToTrack, bool terminalRecord = false)
+        public static async Task<TReturn> TrackAsync<TReturn>(
+            Func<BcdeExecutionTelemetryRecord, CancellationToken, Task<TReturn>> functionToTrack,
+            bool terminalRecord = false,
+            CancellationToken cancellationToken = default)
```

Comment:
`CancellationToken` parameters usually come last. There's [an analyzer rule][1] about this.

[1]: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068

Reply:
That's because this type correlates to `Func<T1, T2, TResult>`[^1]. `T1` and `T2` are the first parameters of a function, and `TResult` is the return type. So a function that takes a `BcdeExecutionTelemetryRecord` as its first parameter and a `CancellationToken` as its second parameter and returns a `Task<TReturn>` satisfies this.

This case is a little bit complicated so imagine a simpler case. If the type is `Func<string, int, bool>`, then something like `var predicated = (str, index) => str.Length == index;` would satisfy this.


[^1]: https://docs.microsoft.com/en-us/dotnet/api/system.func-3?view=net-6.0